test/framework/modules.py

```diff
@@ -750,6 +750,110 @@ def test_exit_code_check(self):
+        self.modtool.load(['OpenMPI/1.6.4-GCC-4.6.4'])
+
+        # default action is to print a clear warning message
+        stderr = check_loaded_modules()
+        patterns = [
+            r"^WARNING: Found one or more non-ignored loaded \(EasyBuild-generated\) modules in current environment:",
+            r"^\* GCC/4.6.4",
+            r"^\* hwloc/1.6.2-GCC-4.6.4",
+            r"^\* OpenMPI/1.6.4-GCC-4.6.4",
+            "This is not recommended since it may affect the installation procedure\(s\) performed by EasyBuild.",
+            "To make EasyBuild allow particular loaded modules, use the --allow-loaded-modules configuration option.",
+            "To specify action to take when loaded modules are detected, use "
+                "--detect-loaded-modules={error,ignore,purge,unload,warn}",
+        ]
+        for pattern in patterns:
+            self.assertTrue(re.search(pattern, stderr, re.M), "Pattern '%s' found in: %s" % (pattern, stderr))
+
+        # reconfigure EasyBuild to ignore loaded modules for GCC & hwloc & error out when loaded modules are detected
+        options = init_config(args=['--allow-loaded-modules=GCC,hwloc', '--detect-loaded-modules=error'])
+        build_options = {
+            'allow_loaded_modules': options.allow_loaded_modules,
+            'detect_loaded_modules': options.detect_loaded_modules,
+        }
+        init_config(build_options=build_options)
+
+        # error mentioning 1 non-ignored module (OpenMPI), both GCC and hwloc loaded modules are ignored
+        error_pattern = "Found one or more non-ignored loaded .* module.*\n\* OpenMPI/1.6.4-GCC-4.6.4"
+        self.assertErrorRegex(EasyBuildError, error_pattern, self.modtool.check_loaded_modules)
+
+        # check for warning message when purge is being run on loaded modules
+        build_options.update({'detect_loaded_modules': 'purge'})
+        init_config(build_options=build_options)
+        expected = "WARNING: Found non-ignored loaded (EasyBuild-generated) modules, "
+        expected += "running 'module purge': OpenMPI/1.6.4-GCC-4.6.4"
+        self.assertEqual(check_loaded_modules(), expected)
+
+        # check for warning message when loaded modules are unloaded
+        self.modtool.load(['OpenMPI/1.6.4-GCC-4.6.4'])
+        build_options.update({'detect_loaded_modules': 'unload'})
+        init_config(build_options=build_options)
+        expected = "WARNING: Unloading non-ignored loaded (EasyBuild-generated) modules: OpenMPI/1.6.4-GCC-4.6.4"
+        self.assertEqual(check_loaded_modules(), expected)
+
+        # when loaded modules are ignored there are no warnings/errors
+        self.modtool.load(['OpenMPI/1.6.4-GCC-4.6.4'])
+        build_options.update({'detect_loaded_modules': 'ignore'})
+        init_config(build_options=build_options)
+        self.assertEqual(check_loaded_modules(), '')
+
+        # error if any $EBROOT* environment variables are defined that don't match a loaded module
```

Comment:
The test is for the `check_loaded_modules` method in `easybuild/tools/modules.py`, which also checks the orphaned `$EBROOT*` env vars, so it makes sense to keep it as a single test?